vault/resource_oci_auth_backend_config.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+func ociAuthBackendConfigResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: ociAuthBackendWrite,
+		ReadContext:   ReadContextWrapper(ociAuthBackendRead),
+		UpdateContext: ociAuthBackendWrite,
+		DeleteContext: ociAuthBackendDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"backend": {
```

Comment:
We have constants for fields like this instead of using raw strings, for example `consts.FieldBackend`. However I think this is is typically called `path` here instead of backend? Please let me know if you think I'm mistaken.